### PR TITLE
[Dialogs] Allow clients to customize title accessibilityLabel.

### PR DESCRIPTION
**Pull request by codeman7**
This change allows clients to customize the title accessibility label.

Related to #8779

### CODE REVIEW

File: components/Dialogs/src/MDCAlertController.h [hunk 1]
```diff
@@ -119,6 +119,11 @@
  */
 @property(nonatomic, nullable, copy) NSString *title;
 
+/**
+ A custom accessibility label for the title.
```
Comment (jverkoey) [inline components/Dialogs/src/MDCAlertController.h:123]: Please describe the behavior when this property is nil. E.g.

```
When nil, the title's accessibilityLabel will be set to the value of @c title.
```

File: components/Dialogs/tests/unit/MDCAlertControllerTests.m [hunk 2]
```diff
@@ -257,6 +257,47 @@ - (void)testAlertControllerSettingTitleAndMessage {
...
+  // Given
+  NSString *title = @"Foo";
+  NSString *titleAccessibilityLabel = @"Bar";
+
+  // When
+  self.alert.title = title;
+  self.alert.titleAccessibilityLabel = titleAccessibilityLabel;
+
+  // Then
+  MDCAlertControllerView *view = (MDCAlertControllerView *)self.alert.view;
```
Comment (jverkoey) [inline components/Dialogs/tests/unit/MDCAlertControllerTests.m]: Consider moving this to the When section to indicate that the timing of the view load is important for this test. Also consider updating the name of the test to indicate that this is testing "AndViewIsNotLoadedYet".

File: components/Dialogs/tests/unit/MDCAlertControllerTests.m [hunk 3]
```diff
@@ -257,6 +257,47 @@ - (void)testAlertControllerSettingTitleAndMessage {
...
+
+  // When
+  self.alert.title = title;
+  MDCAlertControllerView *view = (MDCAlertControllerView *)self.alert.view;
+  self.alert.titleAccessibilityLabel = nil;
+
+  // Then
+  XCTAssertEqualObjects(view.titleLabel.accessibilityLabel, title);
+}
+
```
Comment (jverkoey) [inline components/Dialogs/tests/unit/MDCAlertControllerTests.m:300]: We may want a similar mirror test here for when the view is not loaded yet.

**Review by jverkoey — APPROVED** [review]

File: components/Dialogs/tests/unit/MDCAlertControllerTests.m (hunk 3 above)
Comment (codeman7) [reply components/Dialogs/tests/unit/MDCAlertControllerTests.m:300]: I believe I've addressed this, please let me know if it needs more work.

File: components/Dialogs/tests/unit/MDCAlertControllerTests.m (hunk 2 above)
Comment (codeman7) [reply components/Dialogs/tests/unit/MDCAlertControllerTests.m]: Done

File: components/Dialogs/src/MDCAlertController.h (hunk 1 above)
Comment (codeman7) [reply components/Dialogs/src/MDCAlertController.h:123]: Done

**Pull request merged**